src/libraries/Microsoft.PowerFx.Core/Glue/SimpleResolver.cs

```diff
@@ -48,27 +51,44 @@ internal class SimpleResolver : INameResolver
 
         public IExternalEntity CurrentEntity => null;
 
-        public SimpleResolver(IEnumerable<EnumSymbol> enumSymbols, params TexlFunction[] extraFunctions)
-            : this(extraFunctions)
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SimpleResolver"/> class.
+        /// </summary>
+        /// <param name="config"></param>
+        public SimpleResolver(PowerFxConfig config)            
         {
-            _enums = enumSymbols.ToArray();
-        }
+            _config = config ?? throw new ArgumentNullException(nameof(config));
```

Comment:
Why are we keeping both the config as well as the _library and _enums (which are also stored on the config?)

Reply:
They could be stored in different formats. the _library and _enums are efficient array caches. (might even be something to build as a dictionary).  But at least they've gone from protected to private state now.